### PR TITLE
set juju channel 3.4

### PR DESCRIPTION
**Pull request by zxhdaze**
Setting Juju channel to 3.4, add concurrency.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -83,8 +83,8 @@ unittests:
 	@tox -e unit
 
 functional: build
-	@echo "Executing functional tests in ${CHARM_BUILD_DIR}"
-	@CHARM_LOCATION=${PROJECTPATH} tox -e func
+	@echo "Executing functional tests with ${PROJECTPATH}/${CHARM_NAME}.charm"
+	@cd src && CHARM_LOCATION=${PROJECTPATH} tox -e func -- ${FUNC_ARGS}
```
Comment (jneo8) [inline Makefile:87]: **nitpick:** 

Why we need cd here?

**Review by jneo8 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (zxhdaze) [reply Makefile]: Fixed, thanks for the comment.

**Review by aieri — APPROVED** [review]

**Review by Pjack — APPROVED** [review]

**Review by jneo8 — APPROVED** [review]

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -28,13 +32,12 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - juju-channel: "3.1/stable"
-            command: "make functional"
+          - juju-channel: "3.4/stable"
+            command: "TEST_JUJU3=1 make functional"  # using TEST_JUJU3 due https://github.com/openstack-charmers/zaza/commit/af7eea953dd5d74d3d074fe67b5765dca3911ca6
```
Comment (rgildein) [inline .github/workflows/check.yaml:36]: Now, I'm curious how this works, since you did not set `TEST_JUJU3` or `TEST_*` passenv in tox.

Comment (zxhdaze) [reply .github/workflows/check.yaml:36]: Yeah this is weird, I printed the env and saw it is picked by tox without being set in passenv... I just added it to passenv for now.

**Pull request merged**